DQM/src/DQM/SampleValidation.cxx

```diff
@@ -0,0 +1,130 @@
+#include "DQM/SampleValidation.h"
+#include "SimCore/Event/SimParticle.h"
+#include "SimCore/Event/SimTrackerHit.h"
+#include "Framework/NtupleManager.h"
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+namespace dqm {
+
+  void SampleValidation::configure(framework::config::Parameters& ps) {
+
+    return;
+  }
+
+  void SampleValidation::analyze(const framework::Event& event) {
+
+    //Grab the SimParticle Map and Target Scoring Plane Hits
+    auto targetSPHits(event.getCollection<ldmx::SimTrackerHit>("TargetScoringPlaneHits"));
+    auto particle_map{event.getMap<int, ldmx::SimParticle>("SimParticles")};
+
+    std::vector<int> primary_daughters;
+
+    double hard_thresh = 2500;
+
+    //Loop over all SimParticles
+    for (auto const& it : particle_map) {
```

Comment:
Not something that needs to be fixed but something that is useful is that when iterating over a map you can do 

```C++
for (auto const& [id, particle] : particle_map) { ... }
```
To define the id and particle variables directly without having to do it.second :)